mesonbuild/interpreter/interpreter.py

```diff
@@ -1128,23 +1128,30 @@ def set_backend(self) -> None:
         # The backend is already set when parsing subprojects
         if self.backend is not None:
             return
-        backend = self.coredata.get_option(OptionKey('backend'))
+        backend_name = self.coredata.get_option(OptionKey('backend'))
         from ..backend import backends
-        self.backend = backends.get_backend_from_name(backend, self.build, self)
+
+        if OptionKey('genvslite') in self.user_defined_options.cmd_line_options.keys():
+            # Use of the '--genvslite vsxxxx' option ultimately overrides any '--backend xxx'
+            # option the user may specify.
+            backend_name = self.coredata.get_option(OptionKey('genvslite'))
+            self.backend = backends.get_genvslite_backend(backend_name, self.build, self)
+        else:
+            self.backend = backends.get_backend_from_name(backend_name, self.build, self)
```

Comment:
Why don't we move the other (original) backend_name assignment to the else branch here?

Reply:
Yep. Well spotted.  Done: [28ea078](https://github.com/mesonbuild/meson/pull/11049/commits/28ea0783bc6fe134adb683f2e30cda7d332d3297)